src/azure-cli/azure/cli/command_modules/synapse/operations/accesscontrol.py

```diff
@@ -4,55 +4,67 @@
 # --------------------------------------------------------------------------------------------
 
 from knack.util import CLIError
-from azure.synapse.accesscontrol.models import RoleAssignmentOptions
 from azure.cli.core.util import is_guid
 from azure.graphrbac.models import GraphErrorException
 from msrestazure.azure_exceptions import CloudError
-from .._client_factory import cf_synapse_client_accesscontrol_factory, cf_graph_client_factory
+from .._client_factory import cf_synapse_role_assignments, cf_synapse_role_definitions, cf_graph_client_factory
+from ..constant import ITEM_NAME_MAPPING
 
 
 # List Synapse Role Assignment
-def list_role_assignments(cmd, workspace_name, role=None, assignee=None):
-    # get role id
-    role_id = _resolve_role_id(cmd, role, workspace_name)
-    # get object_id
-    object_id = _resolve_object_id(cmd, assignee, fallback_to_object_id=True)
+def list_role_assignments(cmd, workspace_name, role=None, assignee=None, assignee_object_id=None,
```

Comment:
>assignee=None, assignee_object_id=None, [](start = 58, length = 39)

For what reason we have both `--assignee` and `--assignee-object-id`? Can we just combine them as one parameter?

Reply:
keep same step with 'az role assignment', I also think `--assignee-object-id` can bypass `--assignee` potential issue. 
`--assignee`  parameter needs to invoke graph api which may meeting time delay issue.

Reply:
Make sense to me.

Reply:
I checked that only `az role assignment create` have additional `--assignee-object-id` with `--assignee-principal-type`. `list` and `delete` don't.

May I know why `az synapse role assignment list/delete` need `--assignee-object-id` without `--assignee-principal-type`?

Reply:
1. `--assignee-principal-type` is an optional parameter for  `az role assignment create` , and its code following `az role assignment` 
https://github.com/Azure/azure-cli/blob/50ccb24528eb36529d1432808f3ff385cc4c04aa/src/azure-cli/azure/cli/command_modules/role/custom.py#L146

list and delete api don't support `--assignee-principal-type` parameter

2. `--assignee-object-id` bypass Graph API, it can be sent to server directly without transforming  `assignee` to its object_id. Server only accepted object id in GUID format. `--assignee-principal-type` is not required. 

3. Cx can still using `--assignee 00000000-0000-0000-0000-000000000000`, adding  `--assignee-object-id`  just provide a way for cx to bypass Graph API invocation in case of insufficient privileges and provide assignee more accurately